rs/nns/governance/src/proposals/install_code.rs

```diff
@@ -299,6 +349,38 @@ mod tests {
         );
     }
 
+    #[cfg(feature = "test")]
+    #[test]
+    fn test_upgrade_root_protocol_canister() {
+        let install_code = InstallCode {
+            canister_id: Some(ROOT_CANISTER_ID.get()),
+            wasm_module: Some(vec![1, 2, 3]),
```

Comment:
NC: Future work?

We maybe should do some validation on this field. Specifically, there are ways to detect whether it is gzipped or a raw wasm.

Reply:
Added a comment on the `validate()` to say we could do that. However, I doubt if we want to commit to that (so, not a TODO). I'm pretty sure the current NnsFunction does not validate it.